controllers/clustertopology_controller_compute.go

```diff
@@ -308,6 +332,101 @@ func computeCluster(current *clusterTopologyState, infrastructureCluster, contro
 	return cluster
 }
 
+// computeMachineDeploymentTopologyState computes the desired state for a MachineDeploymentTopology
+func computeMachineDeploymentTopologyState(class *clusterTopologyClass, current *clusterTopologyState, machineDeploymentTopology clusterv1.MachineDeploymentTopology, currentMachineDeployment *clusterv1.MachineDeployment) (machineDeploymentTopologyState, error) {
+	state := machineDeploymentTopologyState{}
+
+	className := machineDeploymentTopology.Class
+	machineDeploymentClass := getMachineDeploymentClass(class, className)
```

Comment:
I think I would prefer handling the case where `getMachineDeploymentClass` is nil. Otherwise we'll get a panic below

Reply:
Done